File: exercism/erlang/space-age/src/space_age.erl
-module(space_age).

-export([ageOn/2, test_version/0]).

secondsOnEarth() ->
	31557600.

rounding() ->
	math:pow(10, 2).

ageOn(mercury, Seconds) ->
	round(((Seconds/0.2408467)/secondsOnEarth()) * rounding())/rounding();

ageOn(venus, Seconds) ->
	round(((Seconds/0.61519726)/secondsOnEarth()) * rounding())/rounding();

ageOn(jupiter, Seconds) ->
	round(((Seconds/11.862615)/secondsOnEarth()) * rounding())/rounding();

ageOn(saturn, Seconds) ->
	round(((Seconds/29.447498)/secondsOnEarth()) * rounding())/rounding();

ageOn(uranus, Seconds) ->
	round(((Seconds/84.016846)/secondsOnEarth()) * rounding())/rounding();

ageOn(neptune, Seconds) ->
	round(((Seconds/164.79132)/secondsOnEarth()) * rounding())/rounding();

ageOn(mars, Seconds) ->
	round(((Seconds/1.8808158)/secondsOnEarth()) * rounding())/rounding();

ageOn(earth, Seconds) ->
	round(((Seconds/1.0)/secondsOnEarth()) * rounding())/rounding().

test_version() -> 1.

File: exercism/elixir/space-age/space_age.exs
defmodule SpaceAge do
  @type planet ::
          :mercury
          | :venus
          | :earth
          | :mars
          | :jupiter
          | :saturn
          | :uranus
          | :neptune

  @doc """
  Return the number of years a person that has lived for 'seconds' seconds is
  aged on 'planet'.
  """
  @planets_conversion  %{:mercury => 0.2408467,
         		 :venus => 0.61519726,
			 :mars => 1.8808158,
			 :jupiter => 11.862615,
			 :saturn => 29.447498,
			 :uranus => 84.016846,
			 :neptune => 164.79132,
			 :earth => 1.0}

  @spec age_on(planet, pos_integer) :: float
  def age_on(planet, seconds) do
	(seconds/Map.get(@planets_conversion, planet)) / 31557600 |> :erlang.float_to_binary([decimals: 2]) |> String.to_float
  end
end
